dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,379 @@
+            response containing account metadata
+        """
+        _log.info(f"[BINANCE] Fetching address metadata for {self.address}")
+        path = f"account/{self.address}"  # params expected inside path string
+        try:
+            response = self._call_node_api(path)
+            return response.json()
+        except exceptions.InterchainConnectionError:
+            _log.warning("[BINANCE] Non 200 response from Binance node.")
+            _log.warning("May have been a 500 Bad Request or a 404 Not Found.")
+
+    def _build_transaction_msg(self, account_response: Dict[str, Any], transaction_payload: str) -> Dict:
+        # cannot send an amount of 0; send funds to yourself and avoid hardcoding a dummy recipient address
+        inputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}
+        outputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}
+        response = self._fetch_account()
+
+        transaction_data = {
+            "account_number": response["account_number"],
+            "sequence": response["sequence"],
+            "from": self.address,
+            "memo": transaction_payload,
+            "msgs": [{"type": "cosmos-sdk/Send", "inputs": [inputs], "outputs": [outputs]}],
+        }
+        return transaction_data
+
+    def sign_transaction(self, built_transaction: Dict[str, Any]) -> str:
+        """Sign a transaction for this network
+        Args:
+            built_transaction: The dictionary of the raw transaction containing:
+                account_number (int): fetched from node endpoint
+                sequence (int): fetched from node endpoint
+                from (str, in hex): address tokens are being sent from
+                memo: (str): arbitrary data [optional]
+                msgs: (list): structured as follows:
+                    inputs = {"address": "FROM_ADDRESS", "coins": [{"amount": INT_VALUE, "denom": "TOKEN_SYMBOL"}]}
+                    outputs = {"address": "TO_ADDRESS", "coins": [{"amount": INT_VALUE, "denom": "TOKEN_SYMBOL"}]}
+                    "msgs": [{"type": "cosmos-sdk/Send", "inputs": [inputs], "outputs": [outputs]}]
+        Returns:
+            String of the signed transaction as base64
+        """
+        try:
+            if self.testnet:
+                tx = TestBnbTransaction.from_obj(built_transaction)
+            else:  # mainnet
+                tx = BnbTransaction.from_obj(built_transaction)
+            _log.info(f"[BINANCE] Signing raw transaction: {tx.signing_json()}")
+            mykeys = keys.DCKeys(pull_keys=False)
+            mykeys.initialize(private_key_string=self.b64_private_key)
+            signature = base64.b64decode(mykeys.make_binance_signature(content=tx.signing_json()))
+            tx.apply_sig(signature, self.pub.serialize(compressed=True))
+            signed_transaction_bytes = tx.encode()
+            # Binance docs lie -- signed_transaction expected to be base64, not hex:
```

Comment:
kill this comment, especially since they've changed the docs anyways

Reply:
Fixed, in both locations.